android/sdl_android/src/main/java/com/smartdevicelink/session/SdlSession.java

```diff
@@ -128,6 +135,20 @@ public void shutdown(String info) {
 
     }
 
+    @Override
+    public void onProtocolMessageReceived(ProtocolMessage msg) {
+        super.onProtocolMessageReceived(msg);
+
+        if (SessionType.RPC.equals(msg.getSessionType())) {
+            RPCMessage rpc = RpcConverter.extractRpc(msg, this.sdlProtocol.getProtocolVersion());
+
+            if (rpc != null && rpc.getFunctionID() != null && rpc.getFunctionID() == REGISTER_APP_INTERFACE) {
+                RegisterAppInterfaceResponse raiResponse = (RegisterAppInterfaceResponse) rpc;
+                VehicleTypeHolder.saveVehicleType(contextWeakReference.get(), raiResponse.getVehicleType(), sdlProtocol.getTransportForSession(SessionType.RPC).getAddress());
+            }
+        }
+    }
+
```

Comment:
@JulianKast Am I correct you are suggesting to remove this caching logic? According to the proposal, we should cache the vehicle type in case we've retrieved it in the RAI as well. Do we still need to remove it?

Reply:
The information is only cached for the Android implementation when used with the router service start up functionality. Therefore the code here, does not belong in this PR. It should only be part of the PR that includes the specific Android changes.